test/start-kubernetes.sh

```diff
@@ -0,0 +1,305 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+
+TEST_DIRECTORY=${TEST_DIRECTORY:-$(dirname $(readlink -f $0))}
+TEST_CONFIG=${TEST_CONFIG:-${TEST_DIRECTORY}/test-config.sh}
+DEPLOYMENT_SUFFIX=${DEPLOYMENT_SUFFIX:-govm}
+CLOUD_USER=${CLOUD_USER:-clear}
+CLUSTER=${CLUSTER:-${CLOUD_USER}-${DEPLOYMENT_SUFFIX}}
+DEPLOYMENT_ID=${DEPLOYMENT_ID:-pmem-csi-${CLUSTER}}
+GOVM_YAML=${GOVM_YAML:-$(mktemp --suffix $DEPLOYMENT_ID.yml)}
+REPO_DIRECTORY=${REPO_DIRECTORY:-$(dirname $TEST_DIRECTORY)}
+TEST_DIRECTORY=${TEST_DIRECTORY:-$(dirname $(readlink -f $0))}
+RESOURCES_DIRECTORY=${RESOURCES_DIRECTORY:-${REPO_DIRECTORY}/_work/resources}
+WORKING_DIRECTORY="${WORKING_DIRECTORY:-${REPO_DIRECTORY}/_work/${CLUSTER}}"
+REGISTRY_NAME=${REGISTRY_NAME:-localhost:5000}
+source ${TEST_CONFIG}
+NODES=( $DEPLOYMENT_ID-master
+	$DEPLOYMENT_ID-worker1
+	$DEPLOYMENT_ID-worker2
+	$DEPLOYMENT_ID-worker3)
+CLOUD="${CLOUD:-true}"
+FLAVOR="${FLAVOR:-medium}"
+SSH_KEY="${SSH_KEY:-${RESOURCES_DIRECTORY}/id_rsa}"
+SSH_PUBLIC_KEY="${SSH_KEY}.pub"
+IMAGE_TAG="${IMAGE_TAG:-canary}"
+EFI="${EFI:-true}"
+KVM_CPU_OPTS="${KVM_CPU_OPTS:-\
+		-m 2G,slots=${TEST_MEM_SLOTS:-2},maxmem=34G -smp 4 \
+		-machine pc,accel=kvm,nvdimm=on}"
+EXTRA_QEMU_OPTS="${EXTRA_QWEMU_OPTS:-\
+		-object memory-backend-file,id=mem1,share=${TEST_PMEM_SHARE:-on},\
+		mem-path=/data/nvdimm0,size=${TEST_PMEM_MEM_SIZE:-32768}M \
+		-device nvdimm,id=nvdimm1,memdev=mem1,label-size=${TEST_PMEM_LABEL_SIZE:-2097152} \
+		-machine pc,nvdimm}"
+CLOUD_USER=${CLOUD_USER:-clear}
+CLOUD_IMAGE="${CLOUD_IMAGE:-$(\
+		curl -s https://download.clearlinux.org/image/latest-images |\
+		awk '/cloud.img/ {print $0}')}"
+IMAGE_URL=${IMAGE_URL:-https://download.clearlinux.org/releases/${CLOUD_IMAGE//[!0-9]/}/clear}
+SSH_TIMEOUT=60
+SSH_ARGS="-oIdentitiesOnly=yes -oStrictHostKeyChecking=no \
+	-oUserKnownHostsFile=/dev/null -oLogLevel=error \
+	-i ${SSH_KEY}"
+CREATE_REGISTRY=${CREATE_REGISTRY:-false}
+CHECK_SIGNED_FILES=${CHECK_SIGNED_FILES:-true}
```

Comment:
Another thing that I just realized: some of the config variables have `TEST_` as prefix (`TEST_INSECURE_REGISTRIES`), others don't (`CHECK_SIGNED_FILES`).

I used the `TEST_` prefix to reduce the risk of conflicts with env variables that a developer might have set in his environment. Does that make sense? Should all variables have that prefix?

Or none of them?

Either way is fine, we just should avoid the inconsistency. For example, this is what I have now in a config file:
```
$ cat test/test-config.d/my-config.sh
TEST_INSECURE_REGISTRIES="xxxxxx.intel.com"
CHECK_SIGNED_FILES=false
```

Reply:
I vote for having TEST_ prefix in related variable names